easybuild/framework/easyconfig/types.py

```diff
@@ -555,7 +555,6 @@ def ensure_iterable_license_specs(specs):
     'docurls': LIST_OF_STRINGS,
     'name': string_type,
     'osdependencies': STRING_OR_TUPLE_LIST,
-    'patches': STRING_OR_TUPLE_LIST,
```

Comment:
This is not the right way forward, ideally this is updated to something like `STRING_OR_TUPLE_OR_DICT_LIST` (which may not exist yet).

@akesandgren Do you have an example use case here?

Reply:
A change is required to allow #3994 to function.

Reply:
With a patch definition of:
```
{'name': ''AmberTools-21_CMake-FlexiBLAS.patch', 'alt_location': 'AmberTools'},
```
you currently get
```
failed: "Expected elements to be of type string, tuple or list, got {'name': 'AmberTools-21_CMake-FlexiBLAS.patch', 'alt_location': 'AmberTools'} (<class 'dict'>)"
```

Reply:
My upcoming Amber PR(s) and https://github.com/easybuilders/easybuild-easyconfigs/pull/16123

The above was what @branfosj and @ocaisa talked about. But @ocaisa didn't have time to create it.

Reply:
But yes, since the tests for patches fails in the test suite we need to have that STRING_OR_TUPLE_OR_DICT_LIST defined

Reply:
Made a squash and a force push to get a clean single commit which might work better...